### PR TITLE
Implementations for geminals response

### LINKED ISSUE #325: Implement Lagrangian response solver for AP1roG
Implement the linear response solvers for AP1roG, from #313.

- [x] I've also written up some notes that should be converted to issues, so I shouldn't forget this.

### LINKED ISSUE #328: Implement CI response equations
In order to implement CI response equations, we will first have to accomplish #326 and #327.

We should note that the previous implementation in #313 contained an error: there should be no contribution from any nuclear effects.

### PR DESCRIPTION
**Pull request by lelemmen**
**Short description**
In this PR, I'll implement changes as I need them for writing the geminals response paper.

**Related issues**
This PR:
- closes #325;
- closes #328.

**Additional context**

**Don't forget:** 
- [x] if new files are created: update the Doxygen file, the collective gqcp.hpp header and the CMake files
- [x] update the documentation on [our website](https://github.com/GQCG/gqcg.github.io) and check if new Python bindings should be added

### CODE REVIEW

**Comment by lelemmen** [comment]
This PR is at risk of getting too large again, which is why I'm requesting an intermediate review.

File: gqcp/src/Processing/Properties/vAP1roGElectricalResponseSolver.cpp [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+/**
+ *  @param vap1rog          the optimal vAP1roG parameters
+ */
+vAP1roGElectricalResponseSolver::vAP1roGElectricalResponseSolver(const QCModel::vAP1roG& vap1rog) :
+    vap1rog (vap1rog)
+{}
+
+
+
```
Comment (xdvriend) [inline gqcp/src/Processing/Properties/vAP1roGElectricalResponseSolver.cpp:39]:
```suggestion


```
I think previously we only used two white lines, even between two different blocks of functions (like constructors and public methods). At least I know it's like that in the two operators I have already implemented. But if I'm wrong, feel free to ignore this suggestion.

File: gqcp/src/Processing/Properties/vAP1roGElectricalResponseSolver.cpp [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        // Place the component in the larger matrix
+        F_p.col(m) = F_p_m.asVector();
+    }
+
+
+    return F_p;
+}
+
+
+
```
Comment (xdvriend) [inline gqcp/src/Processing/Properties/vAP1roGElectricalResponseSolver.cpp:94]:
```suggestion


```
See previous comment.

**Comment by lelemmen** [comment]
@xdvriend Thanks for the review. You're absolutely correct by the way: consistency matters!

**Comment by lelemmen** [comment]
I'm using my admin powers to merge this PR.

**Pull request merged**